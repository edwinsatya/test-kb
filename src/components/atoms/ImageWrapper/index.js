import React from "react";

export default function ImageWrapper({ classProps, dataProps }) {
  if (dataProps) {
    return (
      <div className={`${classProps} w-full`}>
        <img
          className="w-full h-20 md:h-48 xl:h-96"
          src={dataProps.imageUrl}
          alt={dataProps.campaignName}
        />
      </div>
    );
  }

  return (
    <div className={`${classProps} w-full`}>
      <img
        className="w-full h-20 md:h-48 xl:h-96"
        src="https://www.poltekkes-solo.ac.id/cni-content/themes/poltekkes/images/noimage.jpg"
        alt="no-image"
      />
    </div>
  );
}
