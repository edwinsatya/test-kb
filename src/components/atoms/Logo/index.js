import React from "react";

export default function Logo({ classProps }) {
  return (
    <div className={`${classProps} rounded-full`}>
      <img
        className="w-full"
        src="https://assets.kitabisa.cc/images/logos/logogram__ktbs_white.png"
        alt="logo"
      />
    </div>
  );
}
