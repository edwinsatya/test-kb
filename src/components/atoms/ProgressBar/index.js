import React from "react";

export default function ProgressBar({ dataProps }) {
  if (dataProps) {
    return (
      <div className="bg-gray-200 h-[4px] md:h-2 xl:h-3 w-full mb-1 xl:mb-2 rounded-md overflow-hidden">
        <div
          className={`${
            dataProps < 100 ? "bg-blue-500" : "bg-pink-500"
          } h-[4px] md:h-2 xl:h-3`}
          style={{ width: `${dataProps}%` }}
        ></div>
      </div>
    );
  } else {
    return (
      <div className="bg-gray-200 h-[4px] md:h-2 xl:h-3 w-full mb-1 xl:mb-2 rounded-md overflow-hidden">
        <div
          className={`bg-gray-200 h-[4px] md:h-2 xl:h-3`}
          style={{ width: `100%` }}
        ></div>
      </div>
    );
  }
}
