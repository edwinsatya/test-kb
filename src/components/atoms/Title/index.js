import React from "react";

export default function Title({ classProps, title }) {
  return <div className={classProps}>{title}</div>;
}
