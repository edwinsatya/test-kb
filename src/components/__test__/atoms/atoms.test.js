import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../../config/redux/store";
import ImageWrapper from "../../atoms/ImageWrapper";
import InputSelect from "../../atoms/Input/Select";
import Logo from "../../atoms/Logo";
import ProgressBar from "../../atoms/ProgressBar";
import Title from "../../atoms/Title";

const dataProps = {
  campaignName: "Tribute to Seventeen",
  imageUrl:
    "https://kitabisa-userupload-01.s3-ap-southeast-1.amazonaws.com/_production/campaign/97347/31_97347_1545999748_841841_f.png",
  percentage: 0,
  donationReceived: "Rp50.000",
  remainderDay: 0,
  donationGoal: 500000000,
};

describe("ImageWrapper Component", () => {
  it("Should render with no Image", () => {
    let component = renderer.create(<ImageWrapper />);
    expect(component).toMatchSnapshot();
  });

  it("Should render with Image from dataProps", () => {
    let component = renderer.create(<ImageWrapper dataProps={dataProps} />);
    expect(component).toMatchSnapshot();
  });
});

describe("InputSelect Component", () => {
  it("Should render normally", () => {
    let component = renderer.create(
      <Provider store={store}>
        <InputSelect />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});

describe("Logo Component", () => {
  it("Should render normally", () => {
    let component = renderer.create(<Logo />);
    expect(component).toMatchSnapshot();
  });
});

describe("ProgressBar Component", () => {
  it("Should render normally with no dataProps", () => {
    let component = renderer.create(<ProgressBar />);
    expect(component).toMatchSnapshot();
  });

  it("Should render normally with dataProps", () => {
    let component = renderer.create(
      <ProgressBar dataProps={dataProps.percentage} />
    );
    expect(component).toMatchSnapshot();
  });
});

describe("Title Component", () => {
  it("Should render normally with no titleProps", () => {
    let component = renderer.create(<Title />);
    expect(component).toMatchSnapshot();
  });

  it("Should render normally with titleProps", () => {
    let component = renderer.create(<Title title="Terkumpul" />);
    expect(component).toMatchSnapshot();
  });
});
