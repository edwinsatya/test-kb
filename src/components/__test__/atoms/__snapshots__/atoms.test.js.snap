// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageWrapper Component Should render with Image from dataProps 1`] = `
<div
  className="undefined w-full"
>
  <img
    alt="Tribute to Seventeen"
    className="w-full h-20 md:h-48 xl:h-96"
    src="https://kitabisa-userupload-01.s3-ap-southeast-1.amazonaws.com/_production/campaign/97347/31_97347_1545999748_841841_f.png"
  />
</div>
`;

exports[`ImageWrapper Component Should render with no Image 1`] = `
<div
  className="undefined w-full"
>
  <img
    alt="no-image"
    className="w-full h-20 md:h-48 xl:h-96"
    src="https://www.poltekkes-solo.ac.id/cni-content/themes/poltekkes/images/noimage.jpg"
  />
</div>
`;

exports[`InputSelect Component Should render normally 1`] = `
<div>
  <select
    className="border h-6 text-xs px-1 md:h-8 md:text-base md:px-3 xl:h-10 xl:text-lg xl:px-5 border-gray-200 bg-gray-200 text-gray-700 rounded focus:outline-none focus:bg-white focus:border-gray-500"
    defaultValue="goal"
    onChange={[Function]}
  >
    <option
      value="goal"
    >
      Sorting by Donation Goal
    </option>
    <option
      value="day"
    >
      Sorting by Days Remaining
    </option>
  </select>
</div>
`;

exports[`Logo Component Should render normally 1`] = `
<div
  className="undefined rounded-full"
>
  <img
    alt="logo"
    className="w-full"
    src="https://assets.kitabisa.cc/images/logos/logogram__ktbs_white.png"
  />
</div>
`;

exports[`ProgressBar Component Should render normally with dataProps 1`] = `
<div
  className="bg-gray-200 h-[4px] md:h-2 xl:h-3 w-full mb-1 xl:mb-2 rounded-md overflow-hidden"
>
  <div
    className="bg-gray-200 h-[4px] md:h-2 xl:h-3"
    style={
      Object {
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`ProgressBar Component Should render normally with no dataProps 1`] = `
<div
  className="bg-gray-200 h-[4px] md:h-2 xl:h-3 w-full mb-1 xl:mb-2 rounded-md overflow-hidden"
>
  <div
    className="bg-gray-200 h-[4px] md:h-2 xl:h-3"
    style={
      Object {
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`Title Component Should render normally with no titleProps 1`] = `<div />`;

exports[`Title Component Should render normally with titleProps 1`] = `
<div>
  Terkumpul
</div>
`;
